data/korath/korath missions.txt

```diff
@@ -257,3 +270,207 @@ mission "Small Scale Delivery"
 	on complete
 		payment 1000
 		dialog `You find a Korath who seems to have been waiting for you in the spaceport. They greet you by showing their palms, then outstretch their hands as if waiting to receive the package. You hand off the small delivery and they hand you a small payment of <payment>.`
+
+mission "Korath Stormhold Artifacts [1] Coded Message"
+	name "Go to <planet>"
+	description `Head to <planet> to deal with a "security issue."`
+	landing
+	to offer
+		has "event: stormhold korath issue"
+		not "Hai Reveal [B12] Devil-Hide Battle: offered"
+	source
+		government "Republic" "Free Worlds" "Syndicate" "Neutral" "Independent" "Hai" "Hai (Unfettered)" "Pirate"
+		not planet Stormhold
+	destination Stormhold
+	on offer
+		conversation
+			`As soon as you land, your communications console beeps urgently. It's a coded message from <planet> that simply reads, "Security issue. Return to <planet> as soon as possible."`
+			`	The digital signature claims to be of diplomatic origin, and from the burning acorn insignia, it is most likely from the Unfettered.`
```

Comment:
Someone gave the Unfettered private keys for digital signatures that appear to humans as `diplomatic`?
I wonder who else they're sending these messages to.
Wasn't there a big deal about a communications blackout earlier? What happened to that?

Reply:
it's entirely possible that the Unfettered just have their own diplomats and signatures that label themselves as diplomatic. This does just say that the signature claims to be diplomatic, not that it matches authentic diplomatic codes issued elsewhere. In the real world, vast numbers of scammers send out emails claiming to be diplomatic staff on a regular basis. Every once in a while they even fool the people who actually deal with diplomats on a regular basis and should know how to verify that.

Chances are they aren't super concerned about a coms blackout, either.

Reply:
A "digital signature" is a thing irl. If jargon like this is to be used, it ought to be used accurately.
It is not as simple as forging a signature. A scammer sending out an email is unlikely to be able to produce a digital signature that genuinely appears to be from a legitimate source unless they have compromised that legitimate source and acquired their private key, and done so without being detected (else the digital signature produced by that private key would be invalidated).
When checking a digital signature for authenticity, a verified list of digital certificates is used.

> it's entirely possible that the Unfettered just have their own diplomats and signatures that label themselves as diplomatic.

If this were the case, then in order to verify their digital signature, we would have to use their own security certificates, at which point it should be pretty obvious: "The digital signature matches some certificates provided by the Hai; the message is from the Unfettered." Though, I feel this would be an unnecessary level of detail.

> This does just say that the signature claims to be diplomatic, not that it matches authentic diplomatic codes issued elsewhere.

See my earlier comments on what a digital signature.
Here are some other, probably better, explanations:
https://www.adobe.com/sign/digital-signatures.html#:~:text=A%20digital%20signature%20is%20a,document%20and%20can%20be%20verified.
https://support.microsoft.com/en-us/office/digital-signatures-and-certificates-8186cd15-e7ac-4a16-8597-22bd163e8e96
https://www.docusign.co.uk/how-it-works/electronic-signature/digital-signature/digital-signature-faq

> In the real world, vast numbers of scammers send out emails claiming to be diplomatic staff on a regular basis. Every once in a while they even fool the people who actually deal with diplomats on a regular basis and should know how to verify that.

Again, such situations either do not make use of legitimate digital signatures or do so using stolen credentials.
Have the Unfettered stolen these credentials? And they're using them to send *me* a message? And leaving a symbol representing themselves in it? Doesn't really make sense.

> Chances are they aren't super concerned about a coms blackout, either.

Everything here indicates that they are using hte normal human infrastructure to distribute this message. The Hai might not care about hte blackout, but the people in control of that infrastructure do.

Reply:
Thank you for the detailed explanation. That being said, it is *very* common for high security organizations to use cryptographic signatures that do not have a public key distributed outside of their organization, generally specifically because if it's important enough to warrant it, they don't want it leaving their network in the first place. Or they distribute the appropriate key to the recipient ahead of time.

In any case, I don't see a problem here because of the wording:

> The digital signature claims to be of diplomatic origin, and from the burning acorn insignia, it is most likely from the Unfettered.`

It **claims** to be of diplomatic origin. The description doesn't say that it successfully passes any kind of verification. The fact that it ends with "it is most likely from the unfettered." also adds to the element of doubt

I've seen plenty of documents that state that they are digitally signed, but then got printed out and scanned at some point; or just refer to an internal company cryptographic source that my software, being outside the company, can't access. So I can see that it **claims** to be from XYZ corporation, but that it cannot be verified. Yes, both these situations render the digital signature absolutely and completely useless in terms of security or guaranteeing origin. But it still tells me who it *claims* to be from, though.

Lastly, there's no indication of *whose* diplomatic group it originated from, other than the logo. That *also* reinforces the idea that the "diplomatic digital signature" is *not* verified or confirmed, since if it *was* verified, we would know for sure whose diplomatic corps it came from.

And as an aside, while I know that "Digital Signatures" are indeed a formalized thing, I've met an awful lot of people, organizations, and even companies that consider the signature line at the bottom of their email to be their digital signature. They typed their name and info into it, that is, to them, as good as putting pen to paper. Given how much of a hassle it is to verify a signature (not to mention how easy they are to duplicate and forge; and the utter lack of standards or protocols dealing with verifying signatures), to most people the two *are* functionally just as good as each other.

Reply:
> it is very common for high security organizations to use cryptographic signatures that do not have a public key distributed outside of their organization, generally specifically because if it's important enough to warrant it, they don't want it leaving their network in the first place.

If this is the case here, and the player character does not have access to the public key for this digital signature, it is not possible to verify it.

> Or they distribute the appropriate key to the recipient ahead of time.

If this were the case here, the player character would know already who it came from. Either the Unfettered gave it to us to ensure their communications can be verified (in which case, there is no real need for the acorn symbol) or one of the government or military officials involved in this operation gave it to us, in which case they would also presumably tell us it is for the Unfettered, instead of just saying it is "diplomatic" (which would allow any of the individuals with the private key to impersonate any diplomat we think could reasonably send this message.)

> The description doesn't say that it successfully passes any kind of verification.

If it hasn't, that's a fairly important detail to leave out. This narrator is really not very good.

> I've seen plenty of documents that state that they are digitally signed, but then got printed out and scanned at some point; or just refer to an internal company cryptographic source that my software, being outside the company, can't access. So I can see that it claims to be from XYZ corporation, but that it cannot be verified. Yes, both these situations render the digital signature absolutely and completely useless in terms of security or guaranteeing origin. But it still tells me who it claims to be from, though.
> 
> Lastly, there's no indication of whose diplomatic group it originated from, other than the logo. That also reinforces the idea that the "diplomatic digital signature" is not verified or confirmed, since if it was verified, we would know for sure whose diplomatic corps it came from.

Again, we should really be told if the digital signature is determined to be fake, or can't be verified.
Not doing so and just saying "Well, I didn't say it was verified," when it is pointed out that it doesn't really make sense just seems like lazy writing.

> And as an aside, while I know that "Digital Signatures" are indeed a formalized thing, I've met an awful lot of people, organizations, and even companies that consider the signature line at the bottom of their email to be their digital signature. They typed their name and info into it, that is, to them, as good as putting pen to paper. Given how much of a hassle it is to verify a signature (not to mention how easy they are to duplicate and forge; and the utter lack of standards or protocols dealing with verifying signatures), to most people the two are functionally just as good as each other.

Here, we are being told this by the *narrator*. Not a character that probably doesn't understand the jargon themselves.
If the narrator says it is a digital signature, it should be a digital signature, not what people who don't actually know what it means think it means.
More generally, if the narrator is going to use technical jargon, it should be used in a technically correct manner.

Reply:
I suspect that most players can understand the nuance behind "claimed," if they chose to actually think about it. I don't think we need to spell it out for them. In fact, I'd generally say that forcing content creators to hit people in the face with "this is not verified, you do not know who this is from" blunt statements is actually a fairly strong negative to our writing.

That being said, swapping "digital" before "signature" for "block" afterwards should be a reasonable change that removes the reference to the technical definition without having to degrade the text with unnecessary bluntness.

```suggestion
			`	The signature block claims to be of diplomatic origin, and from the burning acorn insignia, it is most likely from the Unfettered.`
```

Reply:
I don't think a digital signature is an appropriate thing to insert nuances like "It might not be from who it says it is," because the whole point of a digital signature is that it is verifiable, easily so in a digitally connected world.

Not using such technical terminology is indeed an improvement, but I would just say `signature`, not `signature block`, what does hte `block` mean here? Are you just trying to link this to a digital signature without using that specific term?

If you want to talk about player's picking up on nuance, though, the thing calling out the Unfettered as the probable source isn't really necessary.

Maybe just drop the idea of a "diplomatic signature" altogether. They clearly already have a calling card.

```suggestion
			`	It is signed with a burning acorn insignia.`
```

Reply:
I havent read all of it but to give context, my idea is that in general the Unfettered dont send info over coms, at least not critical one. Hence why its nice the msg itself doesnt say much.
They wouldnt have much experience writing this kind of stuff, I imagine it was done by Hai sympathizers that came with the fleet and had all clearance needed to make it a Hai certified thing if needed, or they made it on their own but it'd be pretty rough

Reply:
I removed the "likely from the Unfettered" part.

Reply:
> I haven't read all of it but to give context, my idea is that in general the Unfettered dont send info over coms, at least not critical one. Hence why its nice the msg itself doesnt say much.

This isn't a critical mission; it is Choot'k giving you right of first refusal on an exciting fight because he can't do the mission, and wants to live vicariously through you.

Reply:
> I don't think a digital signature is an appropriate thing to insert nuances like "It might not be from who it says it is," because the whole point of a digital signature is that it is verifiable, easily so in a digitally connected world.
> 

And yet, verifying someone is who they say they are is increasingly difficult, to the point I sometimes have trouble convincing people I am who I am.

> Not using such technical terminology is indeed an improvement, but I would just say `signature`, not `signature block`, what does hte `block` mean here? Are you just trying to link this to a digital signature without using that specific term?

I swapped a term that you objected to, as it is a technical term and it was being used in the not-functioning-properly form instead of working properly, for another technical term that is in widespread use and probably will continue to be used for as long as emails and text communications are a thing. See https://en.wikipedia.org/wiki/Signature_block . They often have pictures too, unlike a digital signature, which tends not to.

Reply:
> And yet, verifying someone is who they say they are is increasingly difficult, to the point I sometimes have trouble convincing people I am who I am.

You're not using digital signatures, though.

Reply:
We should not assume the distant future will still be digital.

Reply:
How about IFF? I only skimmed this discussion though

Reply:
> We should not assume the distant future will still be digital.

Then don't use modern day technical jargon.

Reply:
> Then don't use modern day technical jargon.

Any message that has to authoritatively state that it comes from a specific source will have a "signature" or "seal" of some sort. That is a tradition going back thousands of years, and there is no reason to believe it won't exist in a few thousand from now.

It is important to the dialog that Captain `<last>` knows the message is signed by an unknown diplomatic authority with a flaming acorn insignia. If someone can come up with a better way to state that, without removing any of that information, then I am open to changing it.

Reply:
Let's imagine you get this as an email. It is digitally signed. You don't have the public key, so you import it from a keyserver. Now it's verified that the sender created the private key, but not the identity of the sender. You check the email address associated with it and it says "diplomat@gmail.com." That is a digital signature that claims to be of diplomatic origin, but you have no idea if the person whose email is "diplomat@gmail.com" and who created the private key is an actual diplomat.

Reply:
The player has the freedom to ignore the suspicious message, if they want to.